tests/uitools/qml_quick/functional_tests/FloorFilterControllerFuncTest/tst_floorfiltercontroller.qml

```diff
@@ -0,0 +1,215 @@
+        username: credential_username
+        password: credential_password
+    }
+    MapView {
+        id: viewMulti
+        Map {
+            id: mapMulti
+            item: PortalItem {
+                itemId: "49520a67773842f1858602735ef538b5"
+                portal: Portal {
+                    Component.onCompleted: load()
+                    onErrorChanged: {
+                        if (loadStatus === Enums.LoadStatusFailedToLoad) {
+                            console.log(error.message);
+                        }
+                    }
+                    credential: viewerCredentialIndoors
+                    url: "https://indoors.maps.arcgis.com/"
+                }
+            }
+        }
+    }
+
+    MapView {
+        id: viewSingle
+        Map {
+            id: mapSingle
+            item: PortalItem {
+                itemId: "f133a698536f44c8884ad81f80b6cfc7"
+                portal: Portal {
+                    Component.onCompleted: load()
+                    onErrorChanged: {
+                        if (loadStatus === Enums.LoadStatusFailedToLoad) {
+                            console.log(error.message);
+                        }
+                    }
+                    url: "https://www.arcgis.com"
+                }
+            }
+        }
+    }
+
+    // 1 site, 1 facility, 3 levels
+    Component {
+        id: controlSingleComponent
+        FloorFilterController {
+            id: single
+            geoView: viewSingle
+            property SignalSpy spy : SignalSpy {
+                target: single
+                signalName: "onLoaded" // signal outputted from FloorFilter once the floorManager has been loaded
+            }
+        }
+    }
+
+    // more sites, more facilities
+    Component {
+        id: controlMultiComponent
+        FloorFilterController {
+            id: multi
+            geoView: viewMulti
+            property SignalSpy spy : SignalSpy {
+                target: multi
+                signalName: "onLoaded" // signal outputted from FloorFilter once the floorManager has been loaded
+            }
+        }
+    }
+
+
+
+    function loadFloorManagerSingle() {
+        var control = createTemporaryObject(controlSingleComponent, floorFilterControllerFuncTest);
+
+        if(!control.floorManager || control.floorManager.loadStatus !== Enums.LoadStatusLoaded) {
+            control.spy.wait(10000);
+        }
+        compare(control.floorManager.loadStatus, Enums.LoadStatusLoaded);
+        return control;
+    }
+
+    function loadFloorManagerMultiple() {
+        var control = createTemporaryObject(controlMultiComponent, floorFilterControllerFuncTest);
+
+        if(!control.floorManager || control.floorManager.loadStatus !== Enums.LoadStatusLoaded) {
+            control.spy.wait(10000);
+        }
+        compare(control.floorManager.loadStatus, Enums.LoadStatusLoaded);
+        return control;
+    }
+
+    function test_allSitesFacilitySelection() {
```

Comment:
I noticed that if I guarantee the order of execution, it is possible to introduce invisible _bugs_ because the previous test modified some common element used by both tests and being always run one after the other, I will probably never find it out!